sormas-ui/src/main/java/de/symeda/sormas/ui/utils/DeleteRestoreController.java

```diff
@@ -37,47 +47,40 @@ public void restoreSelectedItems(List<String> entityUuids, F entityFacade, CoreE
 			});
 	}
 
-	public void performRestoreSelectedItems(List<String> entityUuids, F entityFacade, CoreEntityRestoreMessages messages, Runnable callback) {
-		if (entityUuids.isEmpty()) {
+	private void performRestoreSelectedItems(List<String> entityUuids, F entityFacade, CoreEntityRestoreMessages messages, Runnable callback) {
+
+		int countNotRestoredEntities = 0;
+		StringBuilder notRestoredEntities = new StringBuilder();
+		for (String selectedRow : entityUuids) {
+			try {
+				entityFacade.restore(selectedRow);
+			} catch (Exception e) {
+				countNotRestoredEntities++;
+				notRestoredEntities.append(selectedRow, 0, 6).append(", ");
+			}
+		}
+		if (notRestoredEntities.length() > 0) {
+			notRestoredEntities = new StringBuilder(" " + notRestoredEntities.substring(0, notRestoredEntities.length() - 2) + ". ");
+		}
+		callback.run();
+		if (countNotRestoredEntities == 0) {
```

Comment:
I think that would be easier to read this method with some refactorings.
- by extracting countNotRestoredEntities == 0 check into a method with a proper naming for ex: allTheEntitiesAreRestored(countNotRestoredEntities) which would return a boolean

- the code from the else branch could go into a separate method for ex: displayNotRestoredEntities(countNotRestoredEntities, notRestoredEntities)  or some similar naming